### PR TITLE
feat: support for adding tag types

### PR DESCRIPTION
**Pull request by ruanyl**
Add tag type to model tag field component, user can select tag type from `number` or `string`

Signed-off-by: Yulong Ruan <ruanyl@amazon.com>### Description

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: public/components/register_model/register_model.types.ts [hunk 1]
```diff
@@ -6,6 +6,7 @@
 export interface Tag {
   key: string;
   value: string;
```
Comment (wanglam) [inline public/components/register_model/register_model.types.ts:8]: Should we use `string | number` for value type? Prefer to use export type Tag = {key: string} & ({value: number, type: "number"} | {value: string, type: "string"}).

File: public/components/register_model/tag_field.tsx [hunk 2]
```diff
@@ -113,20 +122,50 @@ export const ModelTagField = ({
...
+  );
+
+  const tagTypeController = useController({
+    name: `tags.${index}.type` as const,
+    control,
+  });
+
+  useEffect(() => {
+    if (selectedTagGroup) {
+      tagTypeController.field.onChange(selectedTagGroup.type);
```
Comment (wanglam) [inline public/components/register_model/tag_field.tsx:137]: Could we move this effect to tagKey onChange event handler? According [react's suggestion](https://react.dev/learn/you-might-not-need-an-effect#sharing-logic-between-event-handlers).

File: public/components/register_model/register_model.types.ts (hunk 1 above)
Comment (ruanyl) [reply public/components/register_model/register_model.types.ts:8]: The default input value from onChange event of `FieldNumber` is `string`, we would need to convert the value in the component. I'm open to this, do you know what does the api expects?

Comment (wanglam) [reply public/components/register_model/register_model.types.ts:8]: We can change it once API finalized.

**Review by wanglam — APPROVED** [review]

**Pull request merged**